hcf_backend/__init__.py

```diff
@@ -273,8 +273,20 @@ def schedule(self, batch):
         self.logger.info('scheduled %d links' % scheduled)
 
     def _process_hcf_link(self, link, score):
-        link.meta.pop(b'origin_is_frontier', None)
-        hcf_request = {'fp': getattr(link, 'meta', {}).get('hcf_fingerprint', link.url)}
+        link_meta = getattr(link, 'meta', {})
+        link_meta.pop(b'origin_is_frontier', None)
+        scrapy_meta = link_meta.get(b'scrapy_meta', {})
+        hcf_request = (
+            link_meta.get('hcf_request')
+            or scrapy_meta.get('hcf_request')
```

Comment:
I would design the scrapy spiders in a way that they're not aware neither of what happens in Scheduler or it's architecture. For your problem I would use `hcf_priority` key in scrapy_meta and read it in backend code. That's it. If Scheduler is changed then nothing happens, it just skips that parameter.